clang/test/CodeGen/attr-counted-by-pr88931.cpp

```diff
@@ -13,7 +13,10 @@ void init(void * __attribute__((pass_dynamic_object_size(0))));
 // CHECK-LABEL: define dso_local void @_ZN3foo3barC1Ev(
 // CHECK-SAME: ptr noundef nonnull align 4 dereferenceable(1) [[THIS:%.*]]) unnamed_addr #[[ATTR0:[0-9]+]] align 2 {
 // CHECK-NEXT:  entry:
-// CHECK-NEXT:    tail call void @_Z4initPvU25pass_dynamic_object_size0(ptr noundef nonnull align 4 dereferenceable(1) [[THIS]], i64 noundef -1) #[[ATTR2:[0-9]+]]
```

Comment:
`_Z4initPvU25pass_dynamic_object_size0 ` is not shown here anymore. Does it still check what it needs to check?

Reply:
The -O2 flag was removed. That means the `__builtin_object_size` stuff won't be evaluated.

Reply:
Yes, that's what I figured. I guess we still need `-O2` for tests that involve `__builtin_object_size`? @Sirraide WDYT?

Reply:
> Yes, that's what I figured. I guess we still need `-O2` for tests that involve `__builtin_object_size`? @Sirraide WDYT?

No idea candidly. I’m not familiar w/ how that builtin works. I’d recommend moving the tests for this to a separate file in that case.

Reply:
What do you mean put in a separate file? This *is* the separate file. :-)

Reply:
Wait, I didn’t notice that this file has nothing to do w/ `__builtin_counted_by_ref` (or am I missing something here?). Because if not, why did you remove the `-O2` from this file?

Reply:
Essentially, what I meant was that there is no reason to ever write a codegen test w/ optimisation enabled—unless the code that you’re introducing has different behaviour depending on optimisation levels (which iirc isn’t the case for `__builtin_counted_by_ref`, right?).

*Other* tests might still require the `-O2`; and if that is the case, then we generally shouldn’t add tests for something that is invariant under optimisation (wrt how the frontend behaves; the IR would be different of course, but then has nothing to do w/ what Clang emits) to a file that contains tests that *do* care about optimisation levels. I thought at first that this is what you did here, but looking at it again, that doesn’t seem to be the case because there *is* in fact a separate file w/ tests for this pr.

So yeah, my bad if I wasn’t too clear about this ealier.

Reply:
> different behaviour depending on optimisation levels

As in, Clang emits different IR before it even gets to the optimiser.

Reply:
I also don't see why the other tests need to be updated here, except if you have to remove test cases that no longer works with new restrictions. Still, the rest of the tests and file checks should stay the same.

Reply:
I removed those tests from this PR. PTAL.

Reply:
That’s better yeah; they seem completely unrelated, so we shouldn’t really be messing w/ them here.